### PR TITLE
Create synthetic data

### PR DESCRIPTION
**Pull request by jpquast**
Bug fixes and major updates in many functions. Also new tests and a `create_synthetic_data` function.

### CODE REVIEW

File: R/qc_ids.R [hunk 1]
```diff
@@ -1,20 +1,27 @@
-#' Check number of precursor/preptide/protein IDs
+#' Check number of precursor, preptide or protein IDs
```
Comment (dschust-r) [inline R/qc_ids.R:1]: *peptide

File: man/go_enrichment.Rd [hunk 2]
```diff
@@ -24,41 +24,46 @@ go_enrichment(
...
-with an ID that corresponds to the IDs in the data. The content of each element should be a character vector with associated GO terms.}
+\item{go_data}{Optional, a data frame that can be obtained with \code{fetch_go}. If you provide data yourself make sure column
+names for protein ID (db_id) and GO ID (go_id) are the same as for data obtained with \code{fetch_go}.}
 
 \item{plot}{A logical indicating whether the result should be plotted or returned as a table.}
+
+\item{plot_cutoff}{A character vector indicating if the plot should contain the top 10 most significant proteins (p-value or adjusted p-value), 
+or if a significance cutoff should be used to determine the number of GO terms in the plot. This information should be provided with the 
+type first followed by the theshold separated by a space. Example are \code{plot_cutoff = "adj_pval top10"}, \code{plot_cutoff = "pval 0.05"} 
+or \code{plot_cutoff = "adj_pval 0.01"}. The threshold can be choosen freely.}
```
Comment (dschust-r) [inline man/go_enrichment.Rd:46]: *chosen

File: man/qc_ids.Rd [hunk 3]
```diff
@@ -2,12 +2,14 @@
 % Please edit documentation in R/qc_ids.R
 \name{qc_ids}
 \alias{qc_ids}
-\title{Check number of precursor/preptide/protein IDs}
+\title{Check number of precursor, preptide or protein IDs}
```
Comment (dschust-r) [inline man/qc_ids.Rd:5]: *peptide

**Review by dschust-r — CHANGES_REQUESTED** [review]
Added a few comments but the rest looks good to me! Seems like you've been quite busy lately :D

File: R/go_enrichment.R [hunk 4]
```diff
@@ -39,84 +45,153 @@
...
+  enrichment_plot <-
+    {
+      if ("term" %in% colnames(plot_input)) {
+        ggplot2::ggplot(plot_input, ggplot2::aes(stats::reorder(.data$term, .data$neg_log_sig), .data$neg_log_sig, fill = .data$direction))
+      } else {
+        ggplot2::ggplot(plot_input, ggplot2::aes(stats::reorder(.data$go_id, .data$neg_log_sig), .data$neg_log_sig, fill = .data$direction))
+      }
+    } +
     ggplot2::geom_col(col = "black", size = 1.5) +
     ggplot2::scale_fill_manual(values = c(Down = "#56B4E9", Up = "#E76145")) +
```
[the remark this answers is not in the document]
Comment (jpquast) [reply R/go_enrichment.R:186]: I would do it red and blue to symbolise up and down. could also be red and green (even though you do not know that color) or so. But would keep it like this.

File: R/barcode_plot.R [hunk 5]
```diff
@@ -36,7 +36,7 @@
 #' facet = pg_protein_accessions
 #' )
 #' }
-barcode_plot <- function(data, start_position, end_position, protein_length, coverage = NULL, fold_change = NULL, coloring = NULL, protein_id = NULL, facet = NULL, fold_change_cutoff = 2)
+barcode_plot <- function(data, start_position, end_position, protein_length, coverage = NULL, fold_change = NULL, colouring = NULL, protein_id = NULL, facet = NULL, fold_change_cutoff = 2)
 {
```
[the remark this answers is not in the document]
Comment (jpquast) [reply R/barcode_plot.R:40]: you mean in addition to `fold_change_cutoff`? or to color peptides in general according to their p-value? this should already be possible using `colouring`.

[the remark this answers is not in the document]
Comment (jpquast) [reply R/barcode_plot.R:42]: can you check if it works well like this now?

File: R/barcode_plot.R [hunk 6]
```diff
@@ -29,14 +31,14 @@
 #' @examples
 #' \dontrun{
 #' barcode_plot(
-#' fold_change = diff,
```
Comment (dschust-r) [inline R/barcode_plot.R:32]: Just realized that you forgot 'data' in the example

File: R/barcode_plot.R (hunk 5 above)
[the remark this answers is not in the document]
Comment (dschust-r) [reply R/barcode_plot.R:42]: Besides the comment on the example, the function works well now.

**Review by dschust-r — CHANGES_REQUESTED** [review]
Added one small comment to barcode_plot, where in the example you forgot 'data'. Other than that: LGTM

**Review by dschust-r — APPROVED** [review]

**Pull request merged**